pyface/data_view/api.py

```diff
@@ -0,0 +1,35 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+
+# NOTE: this public-facing API is provisional and may change in future
+# minor releases through until Pyface 8
+
+from pyface.data_view.abstract_data_exporter import AbstractDataExporter  # noqa: 401
+from pyface.data_view.abstract_data_model import AbstractDataModel  # noqa: 401
+from pyface.data_view.abstract_value_type import AbstractValueType  # noqa: 401
+from pyface.data_view.data_formats import (  # noqa: 401
+    csv_column_format, csv_format, csv_row_format, from_csv, from_csv_column,
+    from_csv_row, from_json, from_npy, html_format, npy_format,
+    standard_text_format, to_csv, to_csv_column, to_csv_row, to_json, to_npy,
+    table_format, text_column_format, text_format, text_row_format
+)
+from pyface.data_view.data_view_errors import (  # noqa: 401
+    DataViewError, DataViewGetError, DataViewSetError
+)
+from pyface.data_view.data_view_widget import DataViewWidget  # noqa: 401
+from pyface.data_view.data_wrapper import DataWrapper  # noqa: 401
+from pyface.data_view.i_data_view_widget import (  # noqa: 401
+    IDataViewWidget, MDataViewWidget
```

Comment:
If mixin classes are considered part of the public API, then the evaluation on #731 will have to include downstream subclasses we don't know about, which basically mean we can't change them easily. Should we keep these mixins as implementation details and remove them from the `api.py`?

Reply:
Actually, I should remove because they are useless here: no code that actually wants to use them can import them from .api because it will trigger toolkit selection and circular imports.  So having them here will just cause noise.

But they are "public" as far as if anyone were to build a new toolkit implementation then they will probably want to use these, but they are also not going to be used often and not usually by end-users. So we can't complain if we make them "private" (ie. not in `.api`) but people use them (they are _intended_ to be used!).

Also I don't see why this changes anything regarding #731 - all of the mixins are potentially used by 3rd party libraries if they are implementing a new toolkit; it's just that there currently aren't any such libraries that we know of.  But in the discussion re #731, it doesn't change the analysis because these mixins are _always_ being mixed in to HasTraits classes.